### PR TITLE
Initial YAML Build and Azure template.json

### PR DESCRIPTION
**Pull request by mwevill**
Initial YAML Build (Will need updating once DF is created as will trigger from the integration branch not master)
Creates Data Factory Resource.
Enables Diagnostic Logging to the Shared OMS.
Creates KeyVault for holding Connection Strings.

### CODE REVIEW

File: Azure/template.json [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+                    "contentVersion": "1.0.0.0"
+                },
+                "parameters": {
+                    "keyVaultName": {
+                        "value": "[variables('keyVaultName')]"
+                    },
+                    "keyVaultAccessPolicies": {
+                        "value": "[concat(variables('KeyVaultAccessPolicies'), variables('parameterKeyVaultSecretAccessPolicies'))]"
+                    },
+                    "enabledForDiskEncryption": {
```
Comment (chelnak) [inline Azure/template.json]: Why is this enabled?

File: Azure/template.json [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+                    ],
+                    "secrets": [
+                        "Get",
+                        "List",
+                        "Backup"
+                    ]
+                }
+            },
+            {
+                "objectId": "[parameters('diskEncryptionAppRegistrationObjectId')]",
```
Comment (chelnak) [inline Azure/template.json]: What is this used for?

File: Azure/template.json [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    },
+    "variables": {
+        "deploymentUrlBase": "https://raw.githubusercontent.com/SkillsFundingAgency/das-platform-building-blocks/master/templates/",
+        "resourceNamePrefix": "[toLower(concat('das-', parameters('resourceEnvironmentName'),'-', parameters('serviceName')))]",
+        "dataFactoryName": "[concat(variables('resourceNamePrefix'), '-df')]",
+        "keyVaultName": "[concat(variables('resourceNamePrefix'), '-kv')]",
+        "location": "[resourceGroup().location]",
+        "keyVaultAccessPolicies": [
+            {
+                "objectId": "[parameters('backupManagementServiceObjectId')]",
```
Comment (chelnak) [inline Azure/template.json]: What is this used for?

File: Azure/template.json [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        "resourceEnvironmentName": {
+            "type": "string"
+        },
+        "serviceName": {
+            "type": "string"
+        },
+        "logAnalytiecsWorkspace": {
+            "type": "string"
+        },
+        "backupManagementServiceObjectId": {
```
Comment (chelnak) [inline Azure/template.json]: What is this for?

File: Azure/template.json [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+            "type": "string"
+        },
+        "diskEncryptionAppRegistrationObjectId": {
```
Comment (chelnak) [inline Azure/template.json]: What is this for?

File: Azure/template.json [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+            "type": "string"
+        },
+        "keyVaultAccessObjectIds": {
```
Comment (chelnak) [inline Azure/template.json]: Should this `objectid` be coming from the MSI defined below?

File: Azure/template.json [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+    "resources": [
+        {
+            "apiVersion": "2018-06-01",
+            "name": "[variables('dataFactoryName')]",
+            "location": "[variables('location')]",
+            "type": "Microsoft.DataFactory/factories",
+            "identity": {
+                "type": "SystemAssigned"
+            },
+            "properties": {
```
Comment (chelnak) [inline Azure/template.json]: Is this empty properties block needed?

File: Azure/template.json [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+            }
+        },
+        {
+            "type": "microsoft.datafactory/factories/providers/diagnosticsettings",
+            "name": "[concat(variables('dataFactoryName'),'/Microsoft.Insights/diagnostics')]",
+            "location": "West Europe",
+            "apiVersion": "2017-05-01-preview",
+            "properties": {
+                "name": "diagnostics",
+                "storageAccountId": null,
```
Comment (chelnak) [inline Azure/template.json]: Are the `null` property assignments needed?

File: Azure/template.json [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+                "eventHubAuthorizationRuleId": null,
+                "eventHubName": null,
+                "workspaceId": "[parameters('logAnalytiecsWorkspace')]",
+                "logs": [
+                    {
+                        "category": "PipelineRuns",
+                        "enabled": true,
+                        "retentionPolicy": {
+                            "enabled": false,
```
Comment (chelnak) [inline Azure/template.json]: Do we need retention policy block if we are setting it to false/0?

File: Azure/template.json [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+                    "keyVaultName": {
+                        "value": "[variables('keyVaultName')]"
+                    },
+                    "keyVaultAccessPolicies": {
+                        "value": "[concat(variables('KeyVaultAccessPolicies'), variables('parameterKeyVaultSecretAccessPolicies'))]"
+                    },
+                    "enabledForDiskEncryption": {
+                        "value": true
+                    },
+                    "enabledForTemplateDeployment": {
```
Comment (chelnak) [inline Azure/template.json]: Might not need this as we are not pulling secrets from arm template.

File: Azure/template.json (hunk 1 above)
Comment (mwevill) [reply Azure/template.json]: Was copying an existing. Probably not needed

File: Azure/template.json (hunk 3 above)
Comment (mwevill) [reply Azure/template.json]: Updated

File: Azure/template.json (hunk 2 above)
Comment (mwevill) [reply Azure/template.json]: UJpdated template

File: Azure/template.json (hunk 4 above)
Comment (mwevill) [reply Azure/template.json]: Updated template

File: Azure/template.json (hunk 5 above)
Comment (mwevill) [reply Azure/template.json]: Updated Template

File: Azure/template.json (hunk 6 above)
Comment (mwevill) [reply Azure/template.json]: Updated to populate using output but left array as allow us to ADD our AAD group for management

File: Azure/template.json (hunk 7 above)
Comment (mwevill) [reply Azure/template.json]: Removed

File: Azure/template.json (hunk 8 above)
Comment (mwevill) [reply Azure/template.json]: I think so, they are in the example I was using

File: Azure/template.json (hunk 9 above)
Comment (mwevill) [reply Azure/template.json]: I think so, they are in the example I was using

File: Azure/template.json (hunk 10 above)
Comment (mwevill) [reply Azure/template.json]: Disabled

File: Azure/template.json [hunk 11]
```diff
@@ -152,13 +116,13 @@
                         "value": "[variables('keyVaultName')]"
                     },
                     "keyVaultAccessPolicies": {
-                        "value": "[concat(variables('KeyVaultAccessPolicies'), variables('parameterKeyVaultSecretAccessPolicies'))]"
+                        "value": "[concat(reference(concat('Microsoft.DataFactory/factories/', variables('dataFactoryName')), '2018-06-01', 'Full').identity.principalId, variables('parameterKeyVaultSecretAccessPolicies'))]"
                     },
                     "enabledForDiskEncryption": {
-                        "value": true
+                        "value": false
```
Comment (chelnak) [inline Azure/template.json]: I think these should be removed.

File: Azure/template.json (hunk 8 above)
Comment (chelnak) [reply Azure/template.json]: I think they can be removed. Based on https://docs.microsoft.com/en-gb/rest/api/monitor/diagnosticsettings/createorupdate

File: Azure/template.json (hunk 10 above)
Comment (chelnak) [reply Azure/template.json]: These can be removed. Based on https://docs.microsoft.com/en-us/azure/templates/microsoft.keyvault/2018-02-14/vaults#vaultproperties-object

File: Azure/template.json (hunk 11 above)
Comment (mwevill) [reply Azure/template.json]: Removed in latest commit.

File: Azure/template.json (hunk 8 above)
Comment (mwevill) [reply Azure/template.json]: Removed

File: Azure/template.json (hunk 10 above)
Comment (mwevill) [reply Azure/template.json]: Removed

**Pull request merged**